plugins/funind/indfun.ml

```diff
@@ -91,10 +91,22 @@ let functional_induction with_clean c princl pat =
 	if princ_infos.Tactics.farg_in_concl
 	then [c] else []
       in
+      if List.length args + List.length c_list = 0
+      then user_err Pp.(str "Cannot recognize a valid functional scheme" );
       let encoded_pat_as_patlist =
-        List.make (List.length args + List.length c_list - 1) None @ [pat] in
-      List.map2 (fun c pat -> ((None,Ltac_plugin.Tacexpr.ElimOnConstr (fun env sigma -> (sigma,(c,NoBindings)) )),(None,pat),None))
-        (args@c_list) encoded_pat_as_patlist
+        List.make (List.length args + List.length c_list - 1) None @ [pat]
+      in
+      List.map2
+        (fun c pat ->
+          (
+            (None,
+             Ltac_plugin.Tacexpr.ElimOnConstr (fun env sigma -> (sigma,(c,NoBindings)) )
+            ),
+            (None,pat),
+            None)
+        )
```

Comment:
Not sure I understand the indentation of this piece of code...

Reply:
Yes please, use ocp-indent and conform the style prevalent in the codebase.

Reply:
I wouldn't say there is a prevalent style in the code base...
However, I'm interesting to learn what settings you recommend using ocp-indent with. Just the default?
What does it do? I find the official webpage pretty bare. Does it fix the indentation on save? Or does it guarantee a good indentation when you are using the tab key?
It would be very interesting to give such hints in the contributing guide BTW.

Reply:
> I wouldn't say there is a prevalent style in the code base...

Indentation-wise there is some sort of uniformity, at least in the core.

> Just the default?

I find the defaults good.

> What does it do? I find the official webpage pretty bare. Does it fix the indentation on save? Or does it guarantee a good indentation when you are using the tab key?

I use it bound to `tab`.

> It would be very interesting to give such hints in the contributing guide BTW.

I was under the impression it was already discussed in `dev/doc/setup.txt` but I see it is not the case. And in fact, that file is a bit outdated, and in conflict with https://github.com/coq/coq/wiki/DevelSetup

Reply:
Indeed.

Reply:
Done